urlb54e207112_Ghuillaume_CSA_tgz-pJ8-M0_ApplicationJ8/src/M2/Binding.java

```diff
@@ -0,0 +1,73 @@
+package M2;
+
+import M0.Trace;
+
+public class Binding {
+
+	protected String name;
+	protected Interface sideA;
+	protected Interface sideB;
+	
+	
+
+	public Binding(String name) {
+		this.name = name;
+	}
+
+	
+	public final void bind(Interface a, Interface b) {
+		this.sideA = a;
+		this.sideB = b;
+	}
+	
+	
+	/*
+	 * GETTERS AND SETTERS
+	 */
+	
+	public String getName() {
+		return name;
+	}
+
+	public void setName(String name) {
+		this.name = name;
+	}
+
+
+	public final Interface getReceiver(Interface sender) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.